_source/_posts/2022-09-08-react-fastify-postgres.md

```diff
@@ -0,0 +1,917 @@
+```bash
+npx lerna add @fastify/postgres packages/api
+npx lerna add dotenv packages/api
+npx lerna add @okta/jwt-verifier packages/api
+npx lerna add pg packages/api
+```
+
+3. Add Typescript and needed types to the backend repository:
+```bash
+npx lerna add @types/pg packages/api --dev
+npx lerna add typescript packages/api --dev
+```
+
+4. Next, let's create a basic React application using [Create React App](https://github.com/facebook/create-react-app). We'll use the template for Typescript with it as well. In the created `packages/frontend` directory, run:
+```bash
+npx create-react-app . --template typescript
+```
+
+5. Add the frontend Okta dependencies:
+```bash
+npx lerna add @okta/okta-auth-js packages/frontend
+npx lerna add @okta/okta-react packages/frontend
+npx lerna add react-router-dom packages/frontend
+```
+
+6. You can now run `npx lerna bootstrap` to install your package dependencies.
+
+**NOTE:** Our demo repository uses React `^18.2.0` and React Scripts `5.0.1`.
+
+## Setup the PostgreSQL Docker instance
+
+1. In your project's root directory, create a file `docker-compose.yml`:
+{% raw %}
+```yml
+services:
+  postgres:
+    container_name: "nasa-facilities"
+    image: "postgres:latest"
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_USERNAME: "${POSTGRES_USERNAME}"
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      POSTGRES_DB: "${POSTGRES_DB}"
+```
+{% endraw %}
+
+{:start="2"}
+2. Create a `.env` file in the same project root directory and add the following:
+```yml
+POSTGRES_USERNAME=postgres
+POSTGRES_PASSWORD=postgres
+POSTGRES_DB=nasa-facilities
+```
+
+3. To use the Docker configuration we created, run `docker compose up`.
+
+4. There is a provided Postgreql data dump `./nasa-facilities_20200910.sql` in the example repo. You can import it by running the following from the project root directory:
+```bash
+docker exec -i nasa-facilities psql -U postgres nasa-facilities < ./nasa-facilities_20200910.sql
+```
+
+Your Docker instance is now created and running a PostgreSQL database with the restored data.
+
+**NOTE:** The PostgreSQL dump included in the demo repo uses data from [https://www.kaggle.com/datasets/nasa/nasa-facilities](https://www.kaggle.com/datasets/nasa/nasa-facilities).
+
+## Create the API backend app using Fastify
+
+1. Go to your backend repository root directory at `packages/api`.
+
+2. Update `package.json` by adding the following to allow the backend application to compile using Typescript:
+```json
+{
+  "main": "build/index.js",
+  "scripts": {
+    "start": "tsc && node build/index.js"
+  }
+}
+```
+
+3. Initialize a Typescript config file by running:
+```bash
+npx tsc --init
+```
+
+4. Create a `.env` file with the following:
+```yml
+CONNECTION_STRING=PostgreSQL://postgres:postgres@localhost:5432/nasa-facilities
+OKTA_ISSUER=https://{yourOktaDomain}/oauth2/default
+OKTA_CLIENT_ID={yourOktaClientId}
+OKTA_AUDIENCE="api://default"
+```
```

Comment:
Would you mind adding a step or a callout to replace the Okta values with the output from the CLI?